drivers/virtualbox/virtualbox_test.go

```diff
@@ -1,10 +1,207 @@
 package virtualbox
 
 import (
+	"errors"
 	"net"
+	"strings"
 	"testing"
+
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/stretchr/testify/assert"
 )
 
+func TestDriverName(t *testing.T) {
+	driverName := newDriver("default").DriverName()
+
+	assert.Equal(t, "virtualbox", driverName)
+}
+
+func TestDefaultFlags(t *testing.T) {
+	driver, err := create([]string{"test-name"})
+
+	assert.NoError(t, err)
+	assert.Equal(t, driver.CPU, 1)
+	assert.Equal(t, driver.Memory, 1024)
+	assert.Equal(t, driver.DiskSize, 20000)
+	assert.Equal(t, driver.Boot2DockerURL, "")
+	assert.Equal(t, driver.Boot2DockerImportVM, "")
+	assert.Equal(t, driver.HostOnlyCIDR, "192.168.99.1/24")
+	assert.Equal(t, driver.HostOnlyNicType, "82540EM")
+	assert.Equal(t, driver.HostOnlyPromiscMode, "deny")
+	assert.Equal(t, driver.NoShare, false)
+}
+
+func TestCustomFlags(t *testing.T) {
+	driver, err := create([]string{
+		"--virtualbox-cpu-count", "8",
+		"--virtualbox-memory", "4096",
+		"--virtualbox-disk-size", "100000",
+		"--virtualbox-boot2docker-url", "test-boot2docker-url",
+		"--virtualbox-import-boot2docker-vm", "test-boot2docker-vm",
+		"--virtualbox-hostonly-cidr", "192.168.99.1/8",
+		"--virtualbox-hostonly-nictype", "test-nic-type",
+		"--virtualbox-hostonly-nicpromisc", "accept",
+		"--virtualbox-no-share",
+		"test-name"})
+
+	assert.NoError(t, err)
+	assert.Equal(t, driver.CPU, 8)
+	assert.Equal(t, driver.Memory, 4096)
+	assert.Equal(t, driver.DiskSize, 100000)
+	assert.Equal(t, driver.Boot2DockerURL, "test-boot2docker-url")
+	assert.Equal(t, driver.Boot2DockerImportVM, "test-boot2docker-vm")
+	assert.Equal(t, driver.HostOnlyCIDR, "192.168.99.1/8")
+	assert.Equal(t, driver.HostOnlyNicType, "test-nic-type")
+	assert.Equal(t, driver.HostOnlyPromiscMode, "accept")
+	assert.Equal(t, driver.NoShare, true)
+}
+
+func TestSSHHostname(t *testing.T) {
+	hostname, err := newDriver("default").GetSSHHostname()
+
+	assert.Equal(t, "127.0.0.1", hostname)
```

Comment:
I still don't feel particularly strongly that we need tests to check for these values, but if you insist, please move the magic strings (e.g. `127.0.0.1`) to package-level `const` so that we only need to update in one place if we change them.